### PR TITLE
Cleanup toolchain configs

### PR DESCRIPTION
**Pull request by kris701**
* Did a lot of refactoring
* Added the ability to generate a seperate dataset for the classifier to train on

### CODE REVIEW

File: JANGAN/ProjectTools/BaseKerasModelTrainer.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        return (returnTrainSet, returnTestSet)
+
+    def GetDatetimeFromSeconds(self, seconds):
+        return time.strftime("%H:%M:%S", time.gmtime(seconds))
+
+    def PrintStatus(self, iteration, totalIterations, epochTime, epoch):
+        print("No Train loss text implemented.", end="\r")
+
+    def PrintTestStatus(self, iteration, totalIterations, epochTime):
+        print("No Test accuracy text implemented.", end="\r")
```
Comment (jamadaha) [inline JANGAN/ProjectTools/BaseKerasModelTrainer.py:64]: Is it not better to throw an exception?

File: JANGAN/ProjectTools/BaseKerasModelTrainer.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        
+        returnTrainSet = returnTrainSet.shuffle(buffer_size=1024)
+        returnTestSet = returnTestSet.shuffle(buffer_size=1024)
+        return (returnTrainSet, returnTestSet)
+
+    def GetDatetimeFromSeconds(self, seconds):
+        return time.strftime("%H:%M:%S", time.gmtime(seconds))
+
+    def PrintStatus(self, iteration, totalIterations, epochTime, epoch):
+        print("No Train loss text implemented.", end="\r")
```
Comment (jamadaha) [inline JANGAN/ProjectTools/BaseKerasModelTrainer.py:61]: Is it not better to throw an exception?

File: JANGAN/ProjectTools/BaseKerasModelTrainer.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        self.MakeCheckpointRef(ckptPath, self.LatestCheckpointPath)
+
+    def MakeCheckpointRef(self, ckptPathSrc, ckptPathDest):
+        relPath = os.path.relpath(os.path.abspath(ckptPathSrc), os.path.abspath(ckptPathDest))
+        os.makedirs(os.path.dirname(ckptPathDest), exist_ok=True)
+        with open(ckptPathDest, 'w') as f:
+            f.write(f"{ckptPathSrc}")
+
+    def LogData(self, epoch):
+        print("No Data logging implemented")
```
Comment (jamadaha) [inline JANGAN/ProjectTools/BaseKerasModelTrainer.py:83]: Is it not better to throw an exception?

File: JANGAN/ProjectTools/BaseMLModel.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+class BaseMLModel():
+    BatchSize = -1
+    NumberOfChannels = -1
+    NumberOfClasses = -1
+    ImageSize = -1
+    LatentDimension = -1
+    EpochCount = -1
+    RefreshEachStep = -1
+    TensorDatasets = None
+    SaveCheckpoints = True
```
Comment (jamadaha) [inline JANGAN/ProjectTools/BaseMLModel.py:18]: Why is this true by default, as opposed everything that is false?

File: JANGAN/ExperimentQueue/BaseRunConfig.ini [hunk 5]
```diff
@@ -118,4 +158,10 @@ AccuracyThreshold = 0.75
 ## "ExponentialDecay" - Uses https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
 LRScheduler = "Constant"
 # The rate of change for learning
-LearningRateClassifier = 0.0003
\ No newline at end of file
+LearningRateClassifier = 0.0003
+
+[ClassifierOutput]
+# Path of the output logs
+OutputDir = "../../Data/BaseExperiment/{TIMESTAMP}/Logs/"
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/BaseRunConfig.ini:165]: Why is this called OutputDir as opposed to LogDir?

File: JANGAN/ExperimentQueue/BaseRunConfig.ini [hunk 6]
```diff
@@ -43,26 +48,59 @@ LetterOutputFormat = "ZeroIndexed"
 MinimumLetterCount = 0
 MaximumLetterCount = 4000
 
-[CGAN]
+[ClassifierDATAGENERATOR]
```
Comment (HGinnerup) [inline JANGAN/ExperimentQueue/BaseRunConfig.ini:51]: Inconsistent casing

**Review by HGinnerup — CHANGES_REQUESTED** [review]

**Review by HGinnerup — APPROVED** [review]

**Pull request merged**